src/google/protobuf/compiler/csharp/csharp_message.cc

```diff
@@ -132,6 +132,16 @@ void MessageGenerator::Generate(io::Printer* printer) {
   else {
     printer->Print(vars, "pb::IMessage<$class_name$>");
   }
+
+  if (!this->options()->disable_buffer_serialization)
```

Comment:
Inverse and rename to `buffer_serialization`

End result:
Generating buffer serialization code is opt-in. Then choosing to disable it with PROTOBUF_DISABLE_BUFFER_SERIALIZATION is also opt-in.